src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -19,77 +20,108 @@ std::vector<SMALL_RECT> Terminal::_GetSelectionRects() const
-
-        if (_boxSelection || higherCoord.Y == lowerCoord.Y)
-        {
-            selectionRow.Left = std::min(higherCoord.X, lowerCoord.X);
-            selectionRow.Right = std::max(higherCoord.X, lowerCoord.X);
-        }
-        else
-        {
-            selectionRow.Left = (row == higherCoord.Y) ? higherCoord.X : 0;
-            selectionRow.Right = (row == lowerCoord.Y) ? lowerCoord.X : bufferSize.RightInclusive();
-        }
-
-        // expand selection for Double/Triple Click
-        if (_multiClickSelectionMode == SelectionExpansionMode::Word)
-        {
-            const auto cellChar = _buffer->GetCellDataAt(selectionAnchorWithOffset)->Chars();
-            if (_isSingleCellSelection() && _isWordDelimiter(cellChar))
-            {
-                // only highlight the cell if you double click a delimiter
-            }
-            else
-            {
-                selectionRow.Left = _ExpandDoubleClickSelectionLeft({ selectionRow.Left, row }).X;
-                selectionRow.Right = _ExpandDoubleClickSelectionRight({ selectionRow.Right, row }).X;
-            }
-        }
-        else if (_multiClickSelectionMode == SelectionExpansionMode::Line)
-        {
-            selectionRow.Left = 0;
-            selectionRow.Right = bufferSize.RightInclusive();
-        }
-
-        // expand selection for Wide Glyphs
-        selectionRow.Left = _ExpandWideGlyphSelectionLeft(selectionRow.Left, row);
-        selectionRow.Right = _ExpandWideGlyphSelectionRight(selectionRow.Right, row);
+        selectionRow.Left = std::min(higherCoord.X, lowerCoord.X);
+        selectionRow.Right = std::max(higherCoord.X, lowerCoord.X);
+    }
+    else
+    {
+        selectionRow.Left = (row == higherCoord.Y) ? higherCoord.X : _buffer->GetSize().Left();
+        selectionRow.Right = (row == lowerCoord.Y) ? lowerCoord.X : _buffer->GetSize().RightInclusive();
+    }
 
-        selectionArea.emplace_back(selectionRow);
+    return selectionRow;
+}
+
+// Method Description:
+// - Expand the selection row according to selection mode and wide glyphs
+// - this is particularly useful for box selections (ALT + selection)
+// Arguments:
+// - selectionRow: the selection row to be expanded
+// Return Value:
+// - modifies selectionRow's Left and Right values to expand properly
+void Terminal::_ExpandSelectionRow(SMALL_RECT& selectionRow) const
+{
+    const auto row = selectionRow.Top;
+
+    // expand selection for Double/Triple Click
+    if (_multiClickSelectionMode == SelectionExpansionMode::Word)
+    {
+        selectionRow.Left = _ExpandDoubleClickSelectionLeft({ selectionRow.Left, row }).X;
+        selectionRow.Right = _ExpandDoubleClickSelectionRight({ selectionRow.Right, row }).X;
     }
-    return selectionArea;
+    else if (_multiClickSelectionMode == SelectionExpansionMode::Line)
+    {
+        selectionRow.Left = _buffer->GetSize().Left();
+        selectionRow.Right = _buffer->GetSize().RightInclusive();
+    }

```

Comment:
@carlos-zamora @DHowett-MSFT side note: The check for cell here with a do nothing is kind of ugly but it does tell future devs looking at the code that the "cell" case was considered.   One thing I miss in the MSVC C++ compiler is the [-Wswitch-enum](https://clang.llvm.org/docs/DiagnosticsReference.html#wswitch-enum) warning in clang.  If the compiler had this warning then you could make this a switch statement and cover each enum case. (including the Cell's do nothing case) When this flag is used it really helps by giving you compile errors when enum values are added.   It forces you to look at each usage and fix potential errors.

Since the compiler doesn't yet support this we don't have that luxury.   Even so I think adding the do nothing case is still useful for future developers who don't know the history of the code.

Reply:
I did a quick check and MSVC does support this warning - [C4062](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4062?view=vs-2019), but it is off by default.   It is a shame because this warning can be very useful when adding new values to enums/enum classes.